lib/irc/ClientPool.js

```diff
@@ -30,6 +30,33 @@ function ClientPool(ircBridge) {
     }
 }
 
+ClientPool.prototype.killAllClients = function() {
+    var domainList = Object.keys(this._virtualClients);
+    var killList = domainList.map((domain) => {
+        let clients = Object.keys(this._virtualClients[domain].nicks).map(
+            (nick) => this._virtualClients[domain].nicks[nick]
+        );
+
+        clients = clients.concat(
+            Object.keys(this._virtualClients[domain].userIds).map(
+                (userId) => this._virtualClients[domain].userIds[userId]
```

Comment:
Is this `concat` necessary? The nick => Client lookup and user_id => Client lookup should contain the same set of Clients.